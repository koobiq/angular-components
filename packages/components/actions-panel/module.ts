import { NgModule } from '@angular/core';
import { KbqActionsPanel } from './actions-panel';
import { KbqActionsPanelContainer } from './actions-panel-container';

const COMPONENTS = [
    KbqActionsPanelContainer
];

@NgModule({
    imports: COMPONENTS,
    exports: COMPONENTS,
    providers: [KbqActionsPanel]
})
export class KbqActionsPanelModule {}
