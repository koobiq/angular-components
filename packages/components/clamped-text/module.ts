import { NgModule } from '@angular/core';
import { KbqClampedText } from './clamped-text';

const COMPONENTS = [
    KbqClampedText
];

@NgModule({
    imports: COMPONENTS,
    exports: COMPONENTS
})
export class KbqClampedTextModule {}
