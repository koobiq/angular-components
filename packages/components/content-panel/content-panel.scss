.kbq-content-panel {
    display: flex;
    border-radius: var(--kbq-content-panel-border-radius);
    background-color: var(--kbq-content-panel-background-color);
    border: var(--kbq-content-panel-border);
    height: 100%;
    box-sizing: border-box;
    position: relative;

    &.kbq-content-panel_header-with-shadow .kbq-content-panel-header {
        box-shadow: var(--kbq-content-panel-header-overflow-box-shadow);
    }

    &.kbq-content-panel_footer-with-shadow .kbq-content-panel-footer {
        box-shadow: var(--kbq-content-panel-footer-overflow-box-shadow);
    }

    .kbq-content-panel__content {
        display: flex;
        flex: 1;
        flex-direction: column;
        border-top-right-radius: inherit;
        border-bottom-right-radius: inherit;
    }
}
