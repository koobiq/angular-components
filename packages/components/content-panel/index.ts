export * from './content-panel';
export * from './module';
export * from './resizable';
