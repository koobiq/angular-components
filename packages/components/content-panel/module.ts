import { NgModule } from '@angular/core';
import {
    KbqContentPanel,
    KbqContentPanelAside,
    KbqContentPanelBody,
    KbqContentPanelContainer,
    KbqContentPanelFooter,
    KbqContentPanelHeader,
    KbqContentPanelHeaderActions,
    KbqContentPanelHeaderTitle
} from './content-panel';

const COMPONENTS = [
    KbqContentPanelContainer,
    KbqContentPanel,
    KbqContentPanelAside,
    KbqContentPanelHeader,
    KbqContentPanelHeaderTitle,
    KbqContentPanelHeaderActions,
    KbqContentPanelBody,
    KbqContentPanelFooter
];

@NgModule({
    imports: COMPONENTS,
    exports: COMPONENTS
})
export class KbqContentPanelModule {}
