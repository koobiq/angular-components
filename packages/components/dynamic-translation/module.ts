import { NgModule } from '@angular/core';
import { KbqDynamicTranslation, KbqDynamicTranslationHelper, KbqDynamicTranslationSlot } from './dynamic-translation';

const COMPONENTS = [
    KbqDynamicTranslationSlot,
    KbqDynamicTranslationHelper,
    KbqDynamicTranslation
];

@NgModule({
    imports: COMPONENTS,
    exports: COMPONENTS
})
export class KbqDynamicTranslationModule {}
