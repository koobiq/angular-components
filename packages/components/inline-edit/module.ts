import { NgModule } from '@angular/core';
import { KbqInlineEdit, KbqInlineEditEditMode, KbqInlineEditViewMode } from './inline-edit';

const COMPONENTS = [KbqInlineEdit, KbqInlineEditViewMode, KbqInlineEditEditMode];

@NgModule({
    imports: COMPONENTS,
    exports: COMPONENTS
})
export class KbqInlineEditModule {}
