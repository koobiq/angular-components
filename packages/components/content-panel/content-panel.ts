import { animate, state, style, transition, trigger } from '@angular/animations';
import {
    afterNextRender,
    booleanAttribute,
    ChangeDetectionStrategy,
    Component,
    computed,
    contentChild,
    DestroyRef,
    Directive,
    ElementRef,
    inject,
    input,
    numberAttribute,
    output,
    Renderer2,
    signal,
    viewChild,
    ViewEncapsulation
} from '@angular/core';
import { takeUntilDestroyed, toObservable } from '@angular/core/rxjs-interop';
import { KbqButtonModule, KbqButtonStyles } from '@koobiq/components/button';
import { KbqAnimationCurves, KbqAnimationDurations, KbqComponentColors } from '@koobiq/components/core';
import { KbqIconModule } from '@koobiq/components/icon';
import { KbqScrollbar, KbqScrollbarModule } from '@koobiq/components/scrollbar';
import { SizeL } from '@koobiq/design-tokens';
import { KbqResizable, KbqResizer, KbqResizerSizeChangeEvent } from './resizable';

const KBQ_CONTENT_PANEL_CONTAINER_CONTENT_ANIMATION = trigger('contentAnimation', [
    state('false', style({ 'margin-right': 0 })),
    state('true', style({ 'margin-right': '{{ marginRight }}px' }), { params: { marginRight: 0 } }),
    transition('true => false', [animate(`${KbqAnimationDurations.Entering} ${KbqAnimationCurves.AccelerationCurve}`)]),
    transition('false => true', [animate(`${KbqAnimationDurations.Exiting} ${KbqAnimationCurves.DecelerationCurve}`)])
]);

const KBQ_CONTENT_PANEL_CONTAINER_PANEL_ANIMATION = trigger('panelAnimation', [
    transition(':enter', [
        style({ transform: 'translateX(100%)' }),
        animate(
            `${KbqAnimationDurations.Entering} ${KbqAnimationCurves.DecelerationCurve}`,
            style({ transform: 'translateX(0%)' })
        )]),
    transition(':leave', [
        animate(
            `${KbqAnimationDurations.Exiting} ${KbqAnimationCurves.AccelerationCurve}`,
            style({ transform: 'translateX(100%)' })
        )])

]);

@Component({
    standalone: true,
    selector: 'kbq-content-panel-aside',
    exportAs: 'kbqContentPanelAside',
    template: `
        <ng-content />
    `,
    styleUrl: './content-panel-aside.scss',
    host: {
        class: 'kbq-content-panel-aside'
    },
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqContentPanelAside {}

@Directive({
    standalone: true,
    selector: '[kbqContentPanelHeaderTitle]',
    exportAs: 'kbqContentPanelHeaderTitle',
    host: {
        class: 'kbq-content-panel-header-title'
    }
})
export class KbqContentPanelHeaderTitle {}

@Directive({
    standalone: true,
    selector: '[kbqContentPanelHeaderActions]',
    exportAs: 'kbqContentPanelHeaderActions',
    host: {
        class: 'kbq-content-panel-header-actions'
    }
})
export class KbqContentPanelHeaderActions {}

@Component({
    standalone: true,
    imports: [KbqButtonModule, KbqIconModule],
    selector: 'kbq-content-panel-header',
    host: {
        class: 'kbq-content-panel-header'
    },
    template: `
        <div class="kbq-content-panel-header__wrapper">
            <ng-content select="[kbqContentPanelHeaderTitle]" />
            <div class="kbq-content-panel-header__actions">
                <ng-content select="[kbqContentPanelHeaderActions]" />
                @if (!contentPanelContainer.disableClose()) {
                    <button
                        class="kbq-content-panel-header__close-button"
                        [color]="componentColors.Contrast"
                        [kbqStyle]="buttonStyles.Transparent"
                        (click)="contentPanelContainer.close()"
                        kbq-button
                        type="button"
                    >
                        <i kbq-icon="kbq-xmark_16"></i>
                    </button>
                }
            </div>
        </div>
        <ng-content />
    `,
    styleUrl: './content-panel-header.scss',
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqContentPanelHeader {
    /**
     * @docs-private
     */
    protected readonly contentPanelContainer = inject(KbqContentPanelContainer);
    /**
     * @docs-private
     */
    protected readonly buttonStyles = KbqButtonStyles;
    /**
     * @docs-private
     */
    protected readonly componentColors = KbqComponentColors;
}

@Component({
    standalone: true,
    imports: [KbqScrollbarModule],
    selector: 'kbq-content-panel-body',
    exportAs: 'kbqContentPanelBody',
    host: {
        class: 'kbq-content-panel-body'
    },
    template: `
        <kbq-scrollbar class="kbq-content-panel-body__content">
            <ng-content />
        </kbq-scrollbar>
    `,
    styleUrl: './content-panel-body.scss',
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqContentPanelBody {
    /**
     * @docs-private
     */
    readonly scrollbar = viewChild.required(KbqScrollbar);
}

@Component({
    standalone: true,
    selector: 'kbq-content-panel-footer',
    host: {
        class: 'kbq-content-panel-footer'
    },
    template: `
        <ng-content />
    `,
    styleUrl: './content-panel-footer.scss',
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqContentPanelFooter {}

@Component({
    standalone: true,
    selector: 'kbq-content-panel',
    host: {
        class: 'kbq-content-panel'
    },
    template: `
        <ng-content select="kbq-content-panel-aside" />
        <div class="kbq-content-panel__content">
            <ng-content select="kbq-content-panel-header" />
            <ng-content select="kbq-content-panel-body" />
            <ng-content select="kbq-content-panel-footer" />
        </div>
    `,
    styleUrls: [
        './content-panel-tokens.scss',
        './content-panel.scss'
    ],
    encapsulation: ViewEncapsulation.None,
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class KbqContentPanel {
    private readonly destroyRef = inject(DestroyRef);
    private readonly renderer = inject(Renderer2);
    private readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
    private readonly contentPanelBody = contentChild(KbqContentPanelBody);

    constructor() {
        afterNextRender(() => this.handleContentBodyScroll());
    }

    private handleContentBodyScroll(): void {
        const contentPanelBody = this.contentPanelBody();

        if (!contentPanelBody) return;

        contentPanelBody
            .scrollbar()
            .onScroll.pipe(takeUntilDestroyed(this.destroyRef))
            .subscribe(([instance]) => {
                const { scrollTop, scrollHeight, clientHeight } = instance.elements().scrollOffsetElement;

                this.setupContentHeaderShadow(scrollTop >= 1);
                this.setupContentFooterShadow(scrollHeight - (scrollTop + clientHeight) >= 1);
            });
    }

    private setupContentHeaderShadow(shouldShowShadow: boolean): void {
        const className = 'kbq-content-panel_header-with-shadow';

        if (shouldShowShadow) {
            this.renderer.addClass(this.elementRef.nativeElement, className);
        } else {
            this.renderer.removeClass(this.elementRef.nativeElement, className);
        }
    }

    private setupContentFooterShadow(shouldShowShadow: boolean): void {
        const className = 'kbq-content-panel_footer-with-shadow';

        if (shouldShowShadow) {
            this.renderer.addClass(this.elementRef.nativeElement, className);
        } else {
            this.renderer.removeClass(this.elementRef.nativeElement, className);
        }
    }
}

@Component({
    standalone: true,
    imports: [KbqResizable, KbqResizer, KbqScrollbarModule],
    selector: 'kbq-content-panel-container',
    exportAs: 'kbqContentPanelContainer',
    host: {
        class: 'kbq-content-panel-container',
        '[class.kbq-content-panel-container__opened]': 'openedState()',
        '(keydown.escape)': 'handleEscapeKeydown($event)'
    },
    template: `
        <kbq-scrollbar class="kbq-content-panel-container__content" [@contentAnimation]="contentAnimationState()">
            <ng-content />
        </kbq-scrollbar>
        @if (openedState()) {
            <div
                class="kbq-content-panel-container__panel"
                @panelAnimation
                [style.min-width.px]="minWidth()"
                [style.width.px]="widthState()"
                [style.max-width.px]="maxWidth()"
                kbqResizable
            >
                @if (!disableResizer()) {
                    <div
                        class="kbq-content-panel-container__panel-resizer"
                        [kbqResizer]="[-1, 0]"
                        (sizeChange)="handleResizerSizeChange($event)"
                        (dblclick)="handleResizerDBLClick($event)"
                    ></div>
                }
                <ng-content select="kbq-content-panel" />
            </div>
        }
    `,
    styleUrl: './content-panel-container.scss',
    animations: [
        KBQ_CONTENT_PANEL_CONTAINER_CONTENT_ANIMATION,
        KBQ_CONTENT_PANEL_CONTAINER_PANEL_ANIMATION
    ],
    encapsulation: ViewEncapsulation.None,
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class KbqContentPanelContainer {
    /**
     * Whether the content panel is opened.
     *
     * @default false
     */
    readonly opened = input(false, { transform: booleanAttribute });

    /**
     * Emits event when the content panel opened state is changed.
     */
    readonly openedChange = output<boolean>();

    /**
     * Whether the content panel can not be closed by clicking on the close button or pressing the ESCAPE key.
     *
     * @default false
     */
    readonly disableClose = input(false, { transform: booleanAttribute });

    /**
     * Whether the content panel can not be closed by pressing the ESCAPE key.
     */
    readonly disableCloseByEscape = input(this.disableClose(), { transform: booleanAttribute });

    /**
     * Whether the content panel resizer is disabled.
     */
    readonly disableResizer = input(false, { transform: booleanAttribute });

    /**
     * Minimum width of the `KbqContentPanel`.
     *
     * @default 480
     */
    readonly minWidth = input(480, { transform: numberAttribute });

    /**
     * Width of the `KbqContentPanel`.
     *
     * @default 640
     */
    readonly width = input(640, { transform: numberAttribute });

    /**
     * Max width of the `KbqContentPanel`.
     *
     * @default 800
     */
    readonly maxWidth = input(800, { transform: numberAttribute });

    /**
     * @docs-private
     */
    protected readonly openedState = signal(this.opened());

    /**
     * @docs-private
     */
    protected readonly widthState = signal(this.width());

    /**
     * Whether the content panel is opened.
     */
    readonly isOpened = computed(() => this.openedState());

    /**
     * @docs-private
     */
    protected readonly contentAnimationState = computed(() => {
        return {
            value: this.openedState(),
            params: { marginRight: this.widthState() + (parseInt(SizeL) || 16) }
        };
    });

    constructor() {
        // TODO: Should use linked signal
        toObservable(this.opened)
            .pipe(takeUntilDestroyed())
            .subscribe((opened) => {
                this.openedState.set(opened);
            });

        // TODO: Should use linked signal
        toObservable(this.width)
            .pipe(takeUntilDestroyed())
            .subscribe((width) => {
                this.widthState.set(width);
            });
    }

    /**
     * Toggles the content panel opened state.
     */
    toggle(): void {
        this.openedState.update((state) => !state);
        this.openedChange.emit(this.openedState());
    }

    /**
     * Opens the content panel.
     */
    open(): void {
        if (this.openedState()) return;

        this.openedState.set(true);
        this.openedChange.emit(this.openedState());
    }

    /**
     * Closes the content panel.
     */
    close(): void {
        if (!this.openedState()) return;

        this.openedState.set(false);
        this.openedChange.emit(this.openedState());
    }

    /**
     * @docs-private
     */
    protected handleResizerDBLClick(event: MouseEvent): void {
        event.preventDefault();

        this.widthState.set(this.width());
    }

    /**
     * @docs-private
     */
    protected handleResizerSizeChange({ width }: KbqResizerSizeChangeEvent): void {
        if (width > this.maxWidth()) return this.widthState.set(this.maxWidth());
        if (width < this.minWidth()) return this.widthState.set(this.minWidth());

        this.widthState.set(width);
    }

    /**
     * @docs-private
     */
    protected handleEscapeKeydown(event: KeyboardEvent): void {
        if (!this.openedState() || this.disableClose() || this.disableCloseByEscape()) return;

        event.preventDefault();
        event.stopPropagation();

        this.close();
    }
}
