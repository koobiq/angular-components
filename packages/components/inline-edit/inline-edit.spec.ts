import { Component, DebugElement, model, Provider, signal, Type, viewChild } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { KbqContentPanelContainer } from './content-panel';
import { KbqContentPanelModule } from './module';

const createComponent = <T>(component: Type<T>, providers: Provider[] = []): ComponentFixture<T> => {
    TestBed.configureTestingModule({ imports: [component, NoopAnimationsModule], providers });
    const fixture = TestBed.createComponent<T>(component);

    fixture.autoDetectChanges();

    return fixture;
};

const getCloseButtonElement = (debugElement: DebugElement): HTMLButtonElement => {
    return debugElement.nativeElement.querySelector('.kbq-content-panel-header__close-button');
};

const getContentPanelContainerElement = (debugElement: DebugElement): HTMLElement => {
    return debugElement.nativeElement.querySelector('.kbq-content-panel-container')!;
};

const getContentPanelContainerPanelElement = (debugElement: DebugElement): HTMLElement => {
    return debugElement.nativeElement.querySelector('.kbq-content-panel-container__panel')!;
};

const ESCAPE_KEY_EVENT = new KeyboardEvent('keydown', { key: 'Escape' });

@Component({
    standalone: true,
    imports: [KbqContentPanelModule],
    selector: 'test-content-panel',
    template: `
        <kbq-content-panel-container
            [width]="width()"
            [maxWidth]="maxWidth()"
            [minWidth]="minWidth()"
            [disableClose]="disableClose()"
            [disableCloseByEscape]="disableCloseByEscape()"
            [(opened)]="opened"
        >
            <div>test</div>

            <kbq-content-panel>
                <kbq-content-panel-header>
                    <div kbqContentPanelHeaderTitle>Test</div>
                    <div kbqContentPanelHeaderActions>
                        <button>test</button>
                    </div>
                </kbq-content-panel-header>

                <kbq-content-panel-body>
                    @for (_i of paragraphs; track $index) {
                        <p>test</p>
                    }
                </kbq-content-panel-body>

                <kbq-content-panel-footer>
                    <button>test</button>
                </kbq-content-panel-footer>
            </kbq-content-panel>
        </kbq-content-panel-container>
    `
    // changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestContentPanel {
    readonly container = viewChild.required(KbqContentPanelContainer);

    readonly opened = model(false);
    readonly minWidth = signal(300);
    readonly width = signal(400);
    readonly maxWidth = signal(500);
    readonly disableClose = signal(false);
    readonly disableCloseByEscape = signal(false);

    readonly paragraphs = Array.from({ length: 100 });
}

describe(KbqContentPanelModule.name, () => {
    it('should toggle panel by KbqContentPanelContainer instance', () => {
        const { componentInstance } = createComponent(TestContentPanel);

        expect(componentInstance.opened()).toBe(false);

        componentInstance.container().toggle();
        expect(componentInstance.opened()).toBe(true);

        componentInstance.container().toggle();
        expect(componentInstance.opened()).toBe(false);
    });

    it('should open/close panel by KbqContentPanelContainer instance', () => {
        const { componentInstance } = createComponent(TestContentPanel);

        expect(componentInstance.opened()).toBe(false);

        componentInstance.container().close();
        expect(componentInstance.opened()).toBe(false);

        componentInstance.container().open();
        expect(componentInstance.opened()).toBe(true);
    });

    it('should open panel by KbqContentPanelContainer instance', () => {
        const { componentInstance } = createComponent(TestContentPanel);

        expect(componentInstance.opened()).toBe(false);

        componentInstance.container().open();
        expect(componentInstance.opened()).toBe(true);
    });

    it('should open/close panel by two-way binding', () => {
        const { componentInstance } = createComponent(TestContentPanel);

        expect(componentInstance.opened()).toBe(false);

        componentInstance.opened.set(true);
        expect(componentInstance.opened()).toBe(true);

        componentInstance.opened.set(false);
        expect(componentInstance.opened()).toBe(false);
    });

    it('should close panel by close button', () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.container().open();
        fixture.detectChanges();
        expect(componentInstance.opened()).toBe(true);

        getCloseButtonElement(debugElement).click();

        expect(componentInstance.opened()).toBe(false);
    });

    it('should hide close button by disableClose attribute', async () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.disableClose.set(true);
        componentInstance.container().open();
        await fixture.whenStable();
        expect(componentInstance.opened()).toBe(true);

        expect(getCloseButtonElement(debugElement)).toBeNull();
    });

    it('should close panel by pressing ESCAPE key', async () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.container().open();
        await fixture.whenStable();
        expect(componentInstance.opened()).toBe(true);

        getContentPanelContainerElement(debugElement).dispatchEvent(ESCAPE_KEY_EVENT);
        expect(componentInstance.opened()).toBe(false);
    });

    it('should NOT close panel by pressing ESCAPE key when disableClose attribute is provided', async () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.disableClose.set(true);
        componentInstance.container().open();
        await fixture.whenStable();
        expect(componentInstance.opened()).toBe(true);

        getContentPanelContainerElement(debugElement).dispatchEvent(ESCAPE_KEY_EVENT);
        expect(componentInstance.opened()).toBe(true);
    });

    it('should NOT close panel by pressing ESCAPE key when disableCloseByEscape attribute is provided', async () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.disableClose.set(false);
        componentInstance.disableCloseByEscape.set(true);
        componentInstance.container().open();
        await fixture.whenStable();
        expect(componentInstance.opened()).toBe(true);

        getContentPanelContainerElement(debugElement).dispatchEvent(ESCAPE_KEY_EVENT);
        expect(componentInstance.opened()).toBe(true);
    });

    it('should setup width/minWidth/maxWidth', async () => {
        const fixture = createComponent(TestContentPanel);
        const { componentInstance, debugElement } = fixture;

        componentInstance.minWidth.set(222);
        componentInstance.width.set(333);
        componentInstance.maxWidth.set(444);
        componentInstance.container().open();
        await fixture.whenStable();

        const { style } = getContentPanelContainerPanelElement(debugElement);

        expect(style.minWidth).toBe('222px');
        expect(style.width).toBe('333px');
        expect(style.maxWidth).toBe('444px');
    });
});
