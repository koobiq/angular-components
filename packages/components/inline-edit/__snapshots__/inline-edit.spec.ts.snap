// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KbqInlineEdit should add css class when label provided 1`] = `
[
  "kbq-inline-edit",
  "kbq-inline-edit_view",
  "kbq-inline-edit_with-label",
]
`;

exports[`KbqInlineEdit should setup with default parameters 1`] = `
[
  "kbq-inline-edit",
  "kbq-inline-edit_view",
]
`;
